app/models/user.rb

```diff
@@ -1,5 +1,15 @@
 class User < ApplicationRecord
-  has_many :posts
-  has_many :comments
-  has_many :likes
+  has_many :posts, foreign_key: 'author_id', dependent: :destroy
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  before_save :update_posts_counter
+
+  def update_posts_counter
+    self.posts_counter = posts.count
+  end
+
+  def recent_posts(limit: 3)
+    posts.order(created_at: :desc).limit(limit)
+  end
```

Comment:
[OPTIONAL] Nice job making use of the rails active callback😉. Since the method needs to be called after the actions are saved, it will be cool if you use `after_save` instead😉. Also, you should make the method you are using the callback function for private😉. In this case, make the `update_post_counter` private😉.

```suggestion
  after_save :update_posts_counter

  def recent_posts(limit: 3)
    posts.order(created_at: :desc).limit(limit)
  end
   private 
   
  def update_posts_counter
    self.posts_counter = posts.count
  end
```

Reply:
Hi Bro! Thanks for pointing out the mistake. The suggested changes have been implemented. 😃